### PR TITLE
Preparation for a pypi package

### PR DESCRIPTION
**Pull request by bhazelton**
This updates the docs in preparation for making a pypi package including:
- Moving developer instructions (like details on how to do on schema changes) out of the readme and into two dedicated files (for development and site operations) in the docs folder
- Restructuring the installation information. This puts the info about installing the package and dependencies in the readme and the details for setting up the database in a separate file (renamed from "INSTALL.md" to "database_setup.md").
- Adds a developer conda yaml
- updates to the manifest and setup.py for the pypi package
- updating the changelog for the new version
- Adding a pull request template to help us remember all the things that need to be done.

### CODE REVIEW

**Review by mkolopanis — COMMENTED** [review]
This is looking nice. I like all the updated documentation. I have a few suggested changes, mostly  some default branch names.

File: Docs/developer.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+4. Add testing or update code to cover any new code. -- Be sure to `pip install .`
+before the next step (unless you've done a developer install: `pip install -e .`).
+5. Run `alembic revision --autogenerate -m 'version description'` to create a
+new alembic revision file that will reflect the changes to the database schema
+you just introduced. Inspect the resulting file carefully -- alembic's
+autogeneration is very clever but it's certainly not perfect. It tends to make
+more mistakes with table or column alterations than with table creations. Also
+be sure to remove any dropped tables in the upgrade section (and corresponding
+table creation commands in the downgrade portion) -- we have some old tables
+that we no longer use but don't want to drop on site.
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply Docs/developer.md:29]: I don't think there's a way to tell alembic this. I'll bold this sentence.

[the remark this answers is not in the document]
Comment (bhazelton) [reply Docs/developer.md]: oh, wait, there might be a way: https://stackoverflow.com/questions/65184035/alembic-ignore-specific-tables

[the remark this answers is not in the document]
Comment (bhazelton) [reply Docs/developer.md]: also this: https://stackoverflow.com/questions/54978681/ignoring-a-model-when-using-alembic-autogenerate

which may contain hints for skipping tables that have types that don't work with SQLITE...

[the remark this answers is not in the document]
Comment (bhazelton) [reply Docs/developer.md]: I think I'll put these links in an issue, it'll take some work to figure out how to make them work for our case. I don't really want to hold this PR up for that.

[the remark this answers is not in the document]
Comment (mkolopanis) [reply Docs/developer.md]: Sounds good to me. Good find!

**Review by mkolopanis — APPROVED** [review]
I think this is looking good to me. Do we want any other eyes on it?

**Comment by bhazelton** [comment]
@plaplant @david-deboer do you want to take a look or should we just merge this?

**Review by plaplant — APPROVED** [review]
Looks great, thanks for getting things set up!

**Pull request merged**